plasma/morevp_2.0.md

```diff
@@ -0,0 +1,109 @@
+Now, if the child chain works fine, it means that if the chained txs in the cIFE are proven to be non-canonical, the canonical ones are sitting in the correctly propagated valid child chain blocks - users' funds are safe.
+(**TODO** if this is too weak, one can drop the assumption and strengthen the protocol by evicting the non-canonical chained txs in a cIFE).
+
+### Piggybacks of the `super_tx`
+
+Piggybacks have a couple of differences compared to MoreVP IFEs.
+
+Firstly, piggybacks are considered "jointly" for the chaining outputs like `o5` - Bob piggybacks this means that he's piggybacking an output of `tx1` and input of `tx2`.
+This is the same "money" so this makes sense.
+`o5` is then piggybacked with the same meaning, regardless of whether `tx2` is chained/disclosed by `E` or not.
+
+In particular, when Bob piggybacks `o5` before `tx2` is disclosed, Bob's bond is not slashed.
+
+### Remarks
+
+A couple of important notes:
+- **exit priority** - the age of inputs needs to be declared ahead in the step (1/), so it's actually `E.IFE(tx1, inputs_not_younger_than_x)` and priority is based on `x` (in our example flow - that of `o3`).
+During the process of chaining, there can be no inputs "prepended" to the `super_tx` (what happened to `o3`) that are younger than `x`.
+In other words, `E` provides an _a priori_ upper bound on exit priority.
+This is because we cannot easily go about updating the priority queue. If not for `x` declared, we'd been stuck with priority of `o2`, which is the youngest input at time when `E.IFE(tx1...` is done
+- **transaction balance** - i.e. the `sum(inputs)>=sum(outputs)` is really tracked for the whole tx - if all chaining txs satisfy it, so does the entire `super_tx`
+- **unmatched funds on the exchange** - in a variant of the flow, when the chain of txs ends with `(E, D)` holding funds, `D` piggybacks that and `D` will get the payout (see piggybacking)
+- **gas** - cIFE should incur no more gas costs than IFEing all the chained txs, so if we're fine with gas costs of IFE for payment, we should be too with costs of cIFE for settlement (**TODO** - verify this, this is intuition)
+
+Why?
+- in general - simplifies the happy-path, putting the "hard stuff" on the unhappy path - the same reason that made limbo-exit/cancel & MoreVP protocols worthwhile
+- MVP puts us effectively outside the UTXO model - we need some additional conditions that are unverifiable with only UTXO data
+- MVP makes txs enter a shady area in terms of finality - when do we consider a tx done and when we don't?
+In other words, MVP allows to manipulate the doneness of a tx by one of the parties (sender can withhold the confirm sig or receiver can claim lack thereof), and as such is hard to reason about.
+- MVP is further away from MoreVP than MoreVP 2.0 is, with cIFE we're not really introducing anything dramatically new
+- MVP introduces a P2P communication requirement - confirm sig must make its way to the trader when the trade is settled
```

Comment:
The first half is true while the second half is not really an issue. Trader needs to talk to venue anyway.

Reply:
but, the confirm sig travels from venue to trader, in the moment when the trade is settled, requiring the trader to be online (or otherwise complicating their mutual protocol), or am I missing sth?

Reply:
haha, I was imaging the exchange UI would just have a table that shows your order status, and one of the field would be confirm sig.

Reply:
But, this doesn't sound like it's just about the finalization time, isn't this about the ability to secure the exitablity of the funds.

If the confirm sig is missing from the table you've mentioned, to the trader it means that the system is effectively in an unknown state - did the trade happen or not? Is the chain under attack or is the venue acting funny? 

Considering the remedy to this is a forced broadcast of the confirm sig, the trader must force the venue to do that or otherwise take action (exit from the state before the settlement).